packages/headless/src/features/generated-answer/generated-answer-analytics-actions.ts

```diff
@@ -176,26 +190,52 @@ export const logGeneratedAnswerFeedback = (
       if (!generativeQuestionAnsweringId) {
         return null;
       }
-      return client.makeGeneratedAnswerFeedbackSubmit({
-        generativeQuestionAnsweringId,
-        reason: feedback,
-      });
-    },
-    analyticsType: 'Qna.SubmitFeedback',
-    analyticsPayloadBuilder: (state): Qna.SubmitFeedback => {
-      return {
-        answer: {
-          responseId: state.search?.response.searchUid || '',
-          type: RGAType,
-        },
-        feedback: {
-          liked: false,
-          reason: feedback,
-        },
-      };
+      return isGeneratedAnswerFeedbackV2(feedback)
+        ? client.makeGeneratedAnswerFeedbackSubmitV2({
+            generativeQuestionAnsweringId,
+            ...feedback,
+          })
+        : client.makeGeneratedAnswerFeedbackSubmit({
+            generativeQuestionAnsweringId,
+            reason: feedback,
+          });
     },
+    analyticsType: isGeneratedAnswerFeedbackV2(feedback)
+      ? 'Qna.SubmitRgaFeedback'
+      : undefined,
+    analyticsPayloadBuilder: isGeneratedAnswerFeedbackV2(feedback)
+      ? (state): Qna.SubmitRgaFeedback => {
+          const {search} = state;
+          const {response} = search || {};
+          const responseId = response?.searchUid || '';
```

Comment:
🤷 the state is like this all over the code

Reply:
Its because technically, an implementer could try to dispatch any analytics action from any app, even one where the necessary data is not in the store.

ie: Nothing stop someone from calling `dispatch(logGeneratedAnswerFeedback())` from a page that has no gen ai answer component/controller loaded at all. 

It would obviously not do much or be very useful, but it should no crash the app.

As for the syntax, what Etienne propose is easier to read.